plugin/trino-cassandra/src/main/java/io/trino/plugin/cassandra/CassandraClusteringPredicatesExtractor.java

```diff
@@ -65,102 +64,126 @@ private static ClusteringPushDownResult getClusteringKeysSet(List<CassandraColum
             if (domain.isNullAllowed()) {
                 break;
             }
+
+            int currentlyProcessedClusteringColumn = allProcesedClusteringColumns;
             String predicateString = domain.getValues().getValuesProcessor().transform(
                     ranges -> {
-                        List<Object> singleValues = new ArrayList<>();
-                        List<String> rangeConjuncts = new ArrayList<>();
-                        String predicate = null;
-
-                        for (Range range : ranges.getOrderedRanges()) {
-                            if (range.isAll()) {
-                                return null;
-                            }
-                            if (range.isSingleValue()) {
-                                singleValues.add(columnHandle.getCassandraType().toCqlLiteral(range.getSingleValue()));
-                            }
-                            else {
-                                if (!range.isLowUnbounded()) {
-                                    String lowBound = columnHandle.getCassandraType().toCqlLiteral(range.getLowBoundedValue());
-                                    rangeConjuncts.add(format(
-                                            "%s %s %s",
-                                            CassandraCqlUtils.validColumnName(columnHandle.getName()),
-                                            range.isLowInclusive() ? ">=" : ">",
-                                            lowBound));
-                                }
-                                if (!range.isHighUnbounded()) {
-                                    String highBound = columnHandle.getCassandraType().toCqlLiteral(range.getHighBoundedValue());
-                                    rangeConjuncts.add(format(
-                                            "%s %s %s",
-                                            CassandraCqlUtils.validColumnName(columnHandle.getName()),
-                                            range.isHighInclusive() ? "<=" : "<",
-                                            highBound));
-                                }
-                            }
-                        }
-
-                        if (!singleValues.isEmpty() && !rangeConjuncts.isEmpty()) {
+                        if (ranges.getSpan().isAll()) {
                             return null;
```

Comment:
`if (ranges.getSpan().isAll())` then you still could benefit from `ranges.getOrderedRanges().stream().allMatch(Range::isSingleValue)` handling.

(we don't have a logic like that in core yet, but technically `{10, 5, MIN_VALUE, MAX_VALUE}` is set of 4 values, with span being "all")

It seems that you can simply remove

```
if (ranges.getSpan().isAll()) {
  return null;
}
```

block since, this case nicely handled by the following code anyway